### PR TITLE
Fix MacOS build/test failures

### PR DESCRIPTION
**Pull request by kiranchandramohan**
-> Using static_cast in max to match types.
-> Using redirection which works in Mac in a lit test RUN line.
-> Marking a few tests as unsupported on Mac. Issue raised to fix these.
https://github.com/flang-compiler/f18-llvm-project/issues/1338

### CODE REVIEW

**Review by PeixinQiao — APPROVED** [review]
LGTM

File: flang/test/Driver/flang-linker-flags.f90
Comment (banach-space) [inline flang/test/Driver/flang-linker-flags.f90:10]: This could be simplified as:
```
! UNSUPPORTED: darwin, macos
```
See: https://llvm.org/docs/TestingGuide.html#constraining-test-execution

File: flang/test/Fir/complex.fir
Comment (banach-space) [inline flang/test/Fir/complex.fir:8]: This could be simplified as:
```
! UNSUPPORTED: darwin, macos
```
See: https://llvm.org/docs/TestingGuide.html#constraining-test-execution

File: flang/lib/Lower/OpenMP.cpp
Comment (banach-space) [inline flang/lib/Lower/OpenMP.cpp:176]: Why not change the declaration of `loopVarTypeSize` to `std::size_t` instead? This way you could keep the invocation of `std::max` short.

File: flang/test/Driver/flang-linker-flags.f90
Comment (kiranchandramohan) [reply flang/test/Driver/flang-linker-flags.f90]: Done. Good point, will make it work for apple silicon as well.

File: flang/test/Fir/complex.fir
Comment (kiranchandramohan) [reply flang/test/Fir/complex.fir]: Done.

File: flang/lib/Lower/OpenMP.cpp
Comment (kiranchandramohan) [reply flang/lib/Lower/OpenMP.cpp]: Done. Good point.
I was thinking that loopVarTypeSize is used in an MLIR operation which expects a 64 bit quantity. But it does not seem to be so. I have replaced now.

**Comment by kiranchandramohan** [comment]
Addressed review comments. Force pushed changes.

**Review by banach-space — APPROVED** [review]
LGTM, thank you for addressing my comments!

File: flang/test/Fir/complex.fir
Comment (schweitzpgi) [inline flang/test/Fir/complex.fir:8]: What's needed to fix this test, btw?

**Review by schweitzpgi — APPROVED** [review]

File: flang/test/Fir/complex.fir
Comment (banach-space) [reply flang/test/Fir/complex.fir:8]: https://github.com/flang-compiler/f18-llvm-project/issues/1338#issuecomment-992438032

Comment (schweitzpgi) [reply flang/test/Fir/complex.fir:8]: Ah, I see. Thanks, Andrzej. (I expect we'll have to move end-to-end, runnable tests to a different location.)

**Comment by kiranchandramohan** [comment]
Can we merge this patch? Once this is merged I can get the patch that enables the CI ready (https://github.com/flang-compiler/f18-llvm-project/pull/625).

File: flang/test/Fir/complex.fir
Comment (banach-space) [reply flang/test/Fir/complex.fir:8]: > (I expect we'll have to move end-to-end, runnable tests to a different location.)

They are a bit [cross-project](https://github.com/llvm/llvm-project/tree/main/cross-project-tests).

**Review by jeanPerier — APPROVED** [review]
LGTM, thanks

**Pull request merged**
